app/options.go

```diff
@@ -4,25 +4,53 @@
 package app
 
 import (
+	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/store"
 )
 
 type Option func(a *App)
 
+// By default, the app will use a global configuration file. This allows you to override all or part
+// of that configuration.
+//
+// The override parameter must be a *model.Config, func(*model.Config), or func(*model.Config) *model.Config.
+//
+// XXX: Most code will not respect this at the moment. (We need to eliminate utils.Cfg first.)
```

Comment:
This'll be a lot more useful once I eliminate `utils.Cfg`. Does the job for now though (overriding the listen port).